network-api/networkapi/wagtailcustomization/csrf/middleware.py

```diff
@@ -0,0 +1,34 @@
+"""
+Custom CSRF middleware to allow ProductPages to accept POST requests.
+Ever other page should accept CSRF as normal.
+"""
+
+from django.middleware.csrf import CsrfViewMiddleware
+
+from wagtail.core.views import serve
+
+from networkapi.wagtailpages.pagemodels.products import ProductPage
+
+
+class CustomCsrfViewMiddleware(CsrfViewMiddleware):
+
+    def process_view(self, request, callback, callback_args, callback_kwargs):
+
+        if callback == serve and request.method == "POST":
+            # We are visiting a wagtail page. Check if this is a ProductPage
+            # and if so, do not perform any CSRF validation
+            path = callback_args[0].rstrip('/').split('/').pop()   # ie general-percy-product
```

Comment:
Does this have to be callback_args, or does this cover it?
```suggestion
            path = request.path.rstrip('/').split('/').pop()   # ie general-percy-product
```

Reply:
Absolutely we can use the request object. Ive updated this to use the request object, thanks for catching this!